tao_compiler/mlir/xla/ral/ral_context.cc

```diff
@@ -27,12 +29,23 @@
 namespace tao {
 namespace ral {
 
+namespace {
+
+std::atomic<int> nextThreadIdx{0};
+
+}
+
 const char* kRalRecvInput = "ral_recv_input";
 const char* kRalSendOutput = "ral_send_output";
 const char* kRalCudaConst = "ral_const";
 const char* kRalHostConst = "ral_const";
 const char* kRalBitcast = "inc_ref";
 
+int ThreadLocalIndex::Get() {
+  static thread_local int index = nextThreadIdx++;
```

Comment:
Why not use `std::thread::id` directly? https://en.cppreference.com/w/cpp/thread/thread/id

Reply:
`std::thread::id` is an opaque type and thus can not be used to index an array directly. Without the above design, we have to build a hash_map to record the relationship between `std::thread::id` and the array index, which introduces more overhead.